src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -181,7 +194,19 @@ - (void)setDelegate:(id<MTRDeviceDelegate>)delegate queue:(dispatch_queue_t)queu
 
     _weakDelegate = [MTRWeakReference weakReferenceWithObject:delegate];
     _delegateQueue = queue;
-    [self setupSubscription];
+    [self _setupSubscription];
+
+    MTR_LOG_INFO("%@ setDelegate %@", self, delegate);
```

Comment:
Probably worth logging this outside the lock being held, just to minimize how much we do under the lock.